### PR TITLE
[core] stable position of labels at tile borders in tile mode

### PR DESCRIPTION
**Pull request by pozdnyakov**
This PR does two things in Tile mode:
- variable labels, which intercept tile borders, always stick to the first available anchor
- all labels, which intercept tile borders, are placed first

These changes allow to avoid cutting-off labels on tile borders if the variable text placement is enabled.

Fixes https://github.com/mapbox/mapbox-gl-native-team/issues/120

### CODE REVIEW

**Comment by pozdnyakov** [comment]
Can be merged only after :heavy_check_mark: from @mapbox/static-apis

File: src/mbgl/text/placement.cpp [hunk 1]
```diff
@@ -518,6 +553,52 @@ void Placement::placeBucket(
...
+                                                       pitchWithMap,
+                                                       state.getBearing());
+            }
+            bool result = collisionIndex.featureIntersectsTileBorders(
+                symbol.textCollisionFeature, offset, posMatrix, pixelRatio, *tileBorders);
+            sortedCache.insert(std::make_pair(symbol.crossTileID, result));
+            return result;
+        };
+
+        std::stable_sort(sorted.begin(), sorted.end(), [&](const SymbolInstance& a, const SymbolInstance& b) {
```
Comment (alexshalamov) [inline src/mbgl/text/placement.cpp]: capture only lambda instead of &?

**Comment by pozdnyakov** [comment]
Found few errors (visible after we've https://github.com/mapbox/mapbox-gl-native/pull/16068 in place) will apply fix to this PR shortly.

**Comment by riastrad** [comment]
Hooray for the new test runner! 😁

**Comment by pozdnyakov** [comment]
The commits https://github.com/mapbox/mapbox-gl-native/pull/16040/commits/1b55ca875d96af2ad4a5574241a1eddac261b4ff and https://github.com/mapbox/mapbox-gl-native/pull/16040/commits/2ac96794751bff46867b13195f9d24bc47f9897a make the difference comparing to the previous patch set and fix the discovered errors.

File: src/mbgl/text/placement.cpp [hunk 2]
```diff
@@ -518,6 +561,52 @@ void Placement::placeBucket(
...
+            }
+            bool result = collisionIndex.featureIntersectsTileBorders(
+                symbol.textCollisionFeature, offset, posMatrix, pixelRatio, *tileBorders);
+            sortedCache.insert(std::make_pair(symbol.crossTileID, result));
+            return result;
+        };
+
+        std::stable_sort(
+            sorted.begin(), sorted.end(), [&intersectsTileBorder](const SymbolInstance& a, const SymbolInstance& b) {
+                return intersectsTileBorder(a) && !intersectsTileBorder(b);
```
Comment (alexshalamov) [inline src/mbgl/text/placement.cpp]: Does this satisfy [`Compare`](https://en.cppreference.com/w/cpp/named_req/Compare ) requirements?

**Pull request merged**